Reorganize code in Communication

stefano-maggiolo opened the pull request:
This includes the cosmetic changes done to the other task types.

lerks commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[cms/grading/tasktypes/Communication.py, line 68 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ-mf70fxMHjEuKkYu:-LCZ-mf70fxMHjEuKkYv:b-jnayjb) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L68)):*
> ```Python
>     communicate via FIFOs. The manager will read the input, send it (possibly
>     with some modifications) to the user process(es). The user processes, via
>     functions provided by the stub, will read communicate with the manager.
> ```

s/read //

---

*[cms/grading/tasktypes/Communication.py, line 74 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ04f2C6tMqB2NfV8u:-LCZ04f2C6tMqB2NfV8v:b-vmu4d7) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L74)):*
> ```Python
>     The manager reads the input from a file named input.txt and writes the
>     output to output.txt. It receives as argument the names of the fifos:
>     first from and to the first user process, than from and to the second user
> ```

s/than/then/

---

*[cms/grading/tasktypes/Communication.py, line 106 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ0uf7-BIqxsOpssm0:-LCZ0uf7-BIqxsOpssm1:b-26hnjl) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L106)):*
> ```Python
>         self.num_processes = 1
>         if len(self.parameters) > 0:
>             self.num_processes = self.parameters[0]
> ```

Are we really allowing optional parameters now? I feel like this will somehow come to bite us one day...

PS: I just realized it used to be like this before too, just much more implicit... I still don't like it though... do you think anything would break if we changed it?

---

*[cms/grading/tasktypes/Communication.py, line 115 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ1BbF94DStqK7zm_G:-LCZ1BbF94DStqK7zm_H:b-k8o89r) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L115)):*
> ```Python
>             source_ext = language.source_extension
>             source_filenames = ["stub%s" % source_ext]
>             for file_format in submission_format:
> ```

Elsewhere the items of the submission formats are called `element` or `codename`: can we avoid introducing a third name? (same below)

---

*[cms/grading/tasktypes/Communication.py, line 151 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ2KIPC2azWrc49mDv:-LCZ2KIPC2azWrc49mDw:bvr01uu) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L151)):*
> ```Python
>         files_to_get[stub_filename] = job.managers[stub_filename].digest
>         # User's submitted file(s) (copy and add to compilation).
>         for filename, file_ in iteritems(job.files):
> ```

The `filename` here is actually a `element`/`codename`/`file_format`: one day we should try to make naming consistent everywhere...

---

*[cms/grading/tasktypes/Communication.py, line 247 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ4Is44JtYMGZRNQ-P:-LCZ4Is44JtYMGZRNQ-Q:bday2uk) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L247)):*
> ```Python
>             None,
>             allow_dirs=fifo_dir,
>             writable_files=[Communication.OUTPUT_FILENAME],
> ```

What's the use of the output file? I thought the manager was writing outcome and text to stdout/stderr...

PS: ok, I see, it's for when `get_output` is true, hence for user tests... though in tests the manager isn't user-provided, and we still use the admin's one, so maybe we should tell the admin that their manager should log some output? This is confusing...

---


*Comments from [Reviewable](https://reviewable.io/reviews/cms-dev/cms/963#-:-LCZ5tXZ80EoQv7nJPY2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

stefano-maggiolo commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[cms/grading/tasktypes/Communication.py, line 68 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ-mf70fxMHjEuKkYu:-LC_OK466Jh9neYAINuj:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L68)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

s/read //
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Communication.py, line 74 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ04f2C6tMqB2NfV8u:-LC_OKeh4ajPQXa3B9XE:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L74)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

s/than/then/
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Communication.py, line 106 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ0uf7-BIqxsOpssm0:-LC_OLf70A6qm-elYR5e:b-iz2pj2) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L106)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

Are we really allowing optional parameters now? I feel like this will somehow come to bite us one day...

PS: I just realized it used to be like this before too, just much more implicit... I still don't like it though... do you think anything would break if we changed it?
</blockquote></details>

Well, C started as a parameter-free TT, so when we added the num_processes the easiest way was to make it optional. Of course, again, a dict would have been better than a list...

---

*[cms/grading/tasktypes/Communication.py, line 115 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ1BbF94DStqK7zm_G:-LC_OfCF5rdrIY00D-5C:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L115)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

Elsewhere the items of the submission formats are called `element` or `codename`: can we avoid introducing a third name? (same below)
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Communication.py, line 151 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ2KIPC2azWrc49mDv:-LC_Pw9aAkIYpyMEsHnD:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L151)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

The `filename` here is actually a `element`/`codename`/`file_format`: one day we should try to make naming consistent everywhere...
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Communication.py, line 247 at r1](https://reviewable.io/reviews/cms-dev/cms/963#-LCZ4Is44JtYMGZRNQ-P:-LC_PzeADTlg_rq-cUVL:bs3f7wt) ([raw file](https://github.com/cms-dev/cms/blob/640ba45eddb21b3828980da1ebd30cff13eb270a/cms/grading/tasktypes/Communication.py#L247)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

What's the use of the output file? I thought the manager was writing outcome and text to stdout/stderr...

PS: ok, I see, it's for when `get_output` is true, hence for user tests... though in tests the manager isn't user-provided, and we still use the admin's one, so maybe we should tell the admin that their manager should log some output? This is confusing...
</blockquote></details>

Yup, it's confusing. I have also some doubts that we have the correct code for handling Batch user tests with a grader. I kind of wanted to do a check later to see everything works as expected.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cms-dev/cms/963)*
<!-- Sent from Reviewable.io -->

lerks commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cms-dev/cms/963)*
<!-- Sent from Reviewable.io -->
